toolkit/tools/pkg/imagecustomizerlib/cosicommon.go

```diff
@@ -284,3 +284,86 @@ func populateVerityMetadata(source string, verity *Verity) error {
 
 	return nil
 }
+
+// extractPartitionsFromCosi extracts the partition files from a COSI file
+// and returns a list of their paths.
+// It skips directories, metadata.json, and files that are not .zst.
+// It also decompresses the .zst files and writes them to the output directory.
+func extractPartitionsFromCosi(cosiFilePath, outputDir string) ([]string, error) {
```

Comment:
This function is only being used in test code. So, you can probably move it into a `*_test.go` file.